arch/arm/src/lpc214x/lpc214x_serial.c

```diff
@@ -89,7 +90,7 @@ static int  up_attach(struct uart_dev_s *dev);
 static void up_detach(struct uart_dev_s *dev);
 static int  up_interrupt(int irq, void *context, void *arg);
 static int  up_ioctl(struct file *filep, int cmd, unsigned long arg);
-static int  up_receive(struct uart_dev_s *dev, uint32_t *status);
+static int  up_receive(struct uart_dev_s *dev, unsigned int *status);
```

Comment:
I might be a bit out of context (I think you already worked on similar changes in other PRs) but what is the motivation behind this change? I understand that since you know which platform you are in you already now that unsigned int is 32 bit, so no need to explicitly say so. However, aren't we then introducing inconsistent practices in the code? Right now uint32_t is used everywhere and it would be wrong to see some functions use one type and others another type (regardless of actually being the same size).
Also, why is uint32_t a problem? Isn't this definition standard? (defined in stdint.h after all)

Reply:
"Right now uint32_t is used everywhere" is a false assumption.

please see the interface definition here:
https://github.com/apache/incubator-nuttx/blob/c79bda6e4f305c153a117e7ad454ec4e69610d69/include/nuttx/serial/serial.h#L206-L211

also, there is a precedent:
https://github.com/apache/incubator-nuttx/commit/e2aad9c05a085abb9d58b1faa70a631be5e8d89f

Reply:
my motivation is to enable some compiler warnings about types.
unfortunately, our code base is not so consistent in that regard.
that's why i'm making this many changes in this PR and the associated PRs.

Reply:
Well, what I meant is that in some architectures (for example, nRF52) uint32_t is the type used to handle register values and not unsigned int. Similarly, for 16 bit values you can see uint16_t used extensively, instead of unsigned short.

Reply:
are you suggesting that uart_ops_s should be changed? maybe. sorry, i don't volunteer at this point.

Reply:
I was merely asking, trying to understand the context behind this.
I understand now that you are trying to match the declaration of the callback. That sounds OK.

Reply:
And actually I was also curious because I'm used to get a warning from QtCreator's clangbackend whenever I use putreg32, even if I pass uint32_t, it complains that I'm casting an integer to a pointer, which it understands it is not the same size. I always understood this as clangbackend not getting the correct flags, but at the same time I wondered if we should be using uintptr_t in some places instead of uint32_t*. Anyway, just wondering about it, not suggesting any changes.

Reply:
i think it's common for compilers and lint-like tools to warn on non-portable code, even if those types happens to have the same width for the current target.